azavea/usace-program-analysis-geoprocessing
# Add Count Endpoint

## Overview

This endpoint allows us to specify a list of up to 3 layers, and a polygon, and get the count of pixels corresponding to values inside that polygon.

For example, given the following JSON

``` json
{
  "zoom": 0,
  "rasters": [
    "nlcd-2011-30m-epsg5070-0.10.0"
  ],
  "multiPolygon": "{\"type\":\"MultiPolygon\",\"coordinates\":[[[[-75.1626205444336,39.95580659996906],[-75.25531768798828,39.94514735903112],[-75.22785186767578,39.89446035777916],[-75.1461410522461,39.88761144548104],[-75.09309768676758,39.91078961774283],[-75.09464263916016,39.93817189499188],[-75.12039184570312,39.94435771955196],[-75.1626205444336,39.95580659996906]]]]}"
}
```

We get the following response:

``` json
{
    "11": 7569, 
    "21": 5569, 
    "22": 6442, 
    "23": 24881, 
    "24": 39047, 
    "31": 24, 
    "41": 702, 
    "43": 22, 
    "52": 23, 
    "71": 129, 
    "81": 292, 
    "82": 44, 
    "90": 607, 
    "95": 345
}
```

These values can be combined on the client side to get the values for "Wetlands", "Forest", and "Disturbed" as requested in the original card.
## Testing Instructions

Build the JAR file

```
./sbt "project geop" assembly
```

And run it in Spark

```
spark-submit geop/target/scala-2.10/usace-programanalysis-geop-assembly-0.1.0.jar
```

From another window, hit the `/count` endpoint with the `examples/count.json` file. You should see activity in the Spark log, and the following output:

```
curl -H "Content-Type: application/json" --data @examples/count.json http://localhost:8090/count
```

``` json
{
  "23": 24881,
  "11": 7569,
  "90": 607,
  "22": 6442,
  "24": 39047,
  "21": 5569,
  "43": 22,
  "71": 129,
  "82": 44,
  "31": 24,
  "95": 345,
  "81": 292,
  "41": 702,
  "52": 23
}
```

Connects https://github.com/azavea/usace-program-analysis/issues/7
